rasa/train.py

```diff
@@ -149,37 +127,30 @@ def handle_domain_if_not_exists(
     Returns:
         Path of the trained model archive.
     """
-    new_fingerprint = model.model_fingerprint(
-        config, domain, nlu_data_directory, story_directory
-    )
+    new_fingerprint = await model.model_fingerprint(file_importer)
 
-    dialogue_data_not_present = not os.listdir(story_directory)
-    nlu_data_not_present = not os.listdir(nlu_data_directory)
+    dialogue_data = await file_importer.get_story_data()
```

Comment:
Would also rename `dialogue_data` to `stories` here. Also in the print messages.